packages/program-boilerplate/src/conversion.ts

```diff
@@ -1,53 +1,53 @@
-//@ts-check
-
 /**
  * @module conversion
  */
-//typedefs
+
 /**
- * A customField-based conversion rule
- * 
- * @typedef {Object} customFieldConversionRule
- * @property {string} operator A number, boolean or string rule.
- * @property {string} fieldName A number, boolean or string rule.
- * @property {string} value A number, boolean or string rule.
+ * A custom field based conversion rule
  */
+type CustomFieldConversionRule = {
+  operator: number | string
+  fieldName: string,
+  value: RuleValue
+};
 
- /**
- * A event-based conversion rule
- * 
- * @typedef {Object} eventTriggerRule
- * @property {string} operator A number, boolean or string rule.
- * @property {string} eventKey A number, boolean or string rule.
- * @property {string} value A number, boolean or string rule.
+/**
+ * An event-based conversion rule
  */
+type EventTriggerRule = {
+ operator: string
+ eventKey: string
+ value: string
+};
 
 /**
- * @typedef {object} event a key-value pair
- * @property {string} key
- * @property {object} fields An abitrary JSON node
+ * A key-value pair
  */
+type Event = {
+  key: string
+  fields: any
```

Comment:
There is a better type to represent key-value pairs than `any`, see: https://stackoverflow.com/questions/36467469/is-key-value-pair-available-in-typescript/47346860